pkg/kubelet/kubelet.go

```diff
@@ -957,7 +959,17 @@ func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 	}
 
 	// Start volume manager
-	go kl.volumeManager.Run(wait.NeverStop)
+	readyCh := make(chan bool, 1)
+	go kl.volumeManager.Run(readyCh, wait.NeverStop)
+	// Ready channel is set when all the sources are ready
+	go func() {
+		for {
+			if kl.sourcesReady.AllReady() {
+				readyCh <- true
+				break
```

Comment:
Moving the logic to a separate function would also be good clean-up/make it easier to understand.